src/services/services.ts

```diff
@@ -6809,16 +6907,17 @@ namespace ts {
                 }
             }
 
-            function getRelatedSymbol(searchSymbols: Symbol[], referenceSymbol: Symbol, referenceLocation: Node): Symbol {
-                if (searchSymbols.indexOf(referenceSymbol) >= 0) {
-                    return referenceSymbol;
+            function getRelatedSymbol(searchSymbols: Symbol[], referenceSymbol: Symbol, referenceLocation: Node, searchLocationIsConstructor: boolean): Symbol | undefined {
+                if (contains(searchSymbols, referenceSymbol)) {
+                    // If we are searching for constructor uses, they must be 'new' expressions.
+                    return !(searchLocationIsConstructor && !isNewExpressionTarget(referenceLocation)) && referenceSymbol;

```

Comment:
Does this cover the following case?
```
class BaseClass {
	constructor(public p: string) { }
}

function foo(ctor: { new (p: string) }) {
	return new ctor("Test");
}

foo(BaseClass);

}
```

Here the direct use of `BaseClass` is not a new expression target though